browser/brave_origin/brave_origin_service_factory.cc

```diff
@@ -0,0 +1,87 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_origin/brave_origin_service_factory.h"
+
+#include <memory>
+
+#include "base/hash/hash.h"
+#include "base/no_destructor.h"
+#include "base/strings/string_number_conversions.h"
+#include "brave/components/brave_origin/brave_origin_service.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/policy/profile_policy_connector.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/policy/core/common/policy_service.h"
+
+namespace brave_origin {
+
+ProfilePolicyRefreshDelegate::ProfilePolicyRefreshDelegate(
+    policy::PolicyService* policy_service)
+    : policy_service_(policy_service) {}
+
+ProfilePolicyRefreshDelegate::~ProfilePolicyRefreshDelegate() = default;
+
+void ProfilePolicyRefreshDelegate::RefreshPolicies(
+    policy::PolicyFetchReason reason) {
+  if (policy_service_) {
+    policy_service_->RefreshPolicies(base::BindOnce([]() {}), reason);
+  }
+}
+
+// static
+BraveOriginService* BraveOriginServiceFactory::GetForProfile(Profile* profile) {
+  return static_cast<BraveOriginService*>(
+      GetInstance()->GetServiceForBrowserContext(profile, true));
+}
+
+// static
+BraveOriginServiceFactory* BraveOriginServiceFactory::GetInstance() {
+  static base::NoDestructor<BraveOriginServiceFactory> instance;
+  return instance.get();
+}
+
+BraveOriginServiceFactory::BraveOriginServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "BraveOriginService",
+          BrowserContextDependencyManager::GetInstance()) {}
```

Comment:
`ProfileSelections::BuildForRegularAndIncognito()` so this doesn't apply to system/guest profiles